### PR TITLE
Upgrade to Expo SDK 28 along with working example fixed

### PR DESCRIPTION
**Pull request by calebnance**
This is a 2 part PR.

- Upgrade to the latest SDK 28
- Added permissions checking to access CAMERA and CAMERA_ROLL

We shouldn't let first time users shy away from this example repo, if they've never given expo permissions to access their camera, or camera roll... this example will not work at all.

### CODE REVIEW

File: frontend/App.js [hunk 1]
```diff
@@ -11,16 +11,25 @@ import {
   TouchableOpacity,
   View,
 } from 'react-native';
-import Exponent, { Constants, ImagePicker, registerRootComponent } from 'expo';
+import Exponent, { Constants, ImagePicker, Permissions, registerRootComponent } from 'expo';
```
Comment (EvanBacon) [inline frontend/App.js]: Prolly need to change this to `Expo`

File: frontend/App.js [hunk 2]
```diff
@@ -11,16 +11,25 @@ import {
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      image: null,
+      uploading: false,
+    };
+
+    // bind(s)
+    this.allowPermission = this.allowPermission.bind(this);
```
Comment (EvanBacon) [inline frontend/App.js]: arrow function

File: frontend/App.js [hunk 3]
```diff
@@ -49,6 +58,29 @@ export default class App extends React.Component {
     );
   }
 
+  async allowPermission(type) {
```
Comment (EvanBacon) [inline frontend/App.js]: We've changed this logic. Should just be `askAsync` now. If the permission exists, it won't ask again.

File: frontend/App.js [hunk 4]
```diff
@@ -118,21 +150,32 @@ export default class App extends React.Component {
   };
 
   _takePhoto = async () => {
-    let pickerResult = await ImagePicker.launchCameraAsync({
-      allowsEditing: true,
-      aspect: [4, 3],
-    });
+    let permCamera = await this.allowPermission('CAMERA');
+    let permCameraRoll = await this.allowPermission('CAMERA_ROLL');
```
Comment (EvanBacon) [inline frontend/App.js]: Permission consts e.g: `Permissions.CAMERA`

File: frontend/App.js [hunk 5]
```diff
@@ -118,21 +150,32 @@ export default class App extends React.Component {
...
+      this._handleImagePicked(pickerResult);
+    }
   };
 
   _pickImage = async () => {
-    let pickerResult = await ImagePicker.launchImageLibraryAsync({
-      allowsEditing: true,
-      aspect: [4, 3],
-    });
+    let permCameraRoll = await this.allowPermission('CAMERA_ROLL');
```
Comment (EvanBacon) [inline frontend/App.js]: Permission consts

File: frontend/App.js (hunk 2 above)
Comment (EvanBacon) [reply frontend/App.js]: After the arrow function is added,  `this.state = {}` should move to the outer scope. `constructor` function should be deleted.

**Review by EvanBacon — CHANGES_REQUESTED** [review]

File: frontend/App.js (hunk 1 above)
Comment (calebnance) [reply frontend/App.js]: this has been updated

File: frontend/App.js (hunk 2 above)
Comment (calebnance) [reply frontend/App.js]: removed unnecessary `constructor`

File: frontend/App.js (hunk 3 above)
Comment (calebnance) [reply frontend/App.js]: I am going to have to push back on this, I couldn't get it to fire on iOS doing your suggested method. Take a look whenever you have time. 

I am using this as a reusable function `await this.allowPermission('CAMERA')`
Which then uses the suggested const: `Permissions.CAMERA` depending on what type of permission is requested.

Comment (EvanBacon) [reply frontend/App.js]: It seems to work as I had expected: https://snack.expo.io/@bacon/image-upload-s3 
Am I missing something?

Comment (calebnance) [reply frontend/App.js]: Oh my word, I am so sorry, I apologize!

I was not destructuring when i tried to do your suggested solution and my helper function was handling the 'granted' to be a true or false. That was 100% my bad, so so sorry. I've fixed it and pushed to this PR.

**Pull request merged**